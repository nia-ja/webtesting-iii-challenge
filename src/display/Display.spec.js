// Test away!

// ### Display Component
// - displays if gate is open/closed and if it is locked/unlocked
// - displays 'Closed' if the `closed` prop is `true` and 'Open' if otherwise
// - displays 'Locked' if the `locked` prop is `true` and 'Unlocked' if otherwise
// - when `locked` or `closed` use the `red-led` class
// - when `unlocked` or `open` use the `green-led` class

import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/react/cleanup-after-each';
import "jest-dom/extend-expect";

import Display from './Display';

describe('<Display />', () => {
    it('renders without crashing', () => {
        render(<Display />);
    });

    it('matches snapshot', () => {
      const tree = renderer.create(<Display />); 
      expect(tree.toJSON()).toMatchSnapshot();
    });
});