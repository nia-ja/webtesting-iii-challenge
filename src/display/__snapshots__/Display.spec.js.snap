// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="lock-unlock"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="open-close"
  >
    Open
  </div>
</div>
`;
